### DISCUSSION TITLE
71. 인덱스 사용 시 GROUP BY와 ORDER BY의 성능이 향상되는 이유를 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#73] [General]
인덱스 사용 시 GROUP BY와 ORDER BY의 성능이 향상되는 이유를 설명해 주세요.

**Comment by eun-byeol** [comment]
인덱스의 `정렬된 데이터 구조를 활용`할 수 있어 읽기 성능이 향상됩니다.

GROUP BY는 그룹화를 위해 정렬이 필요한데, 인덱스를 활용하면 정렬 연산이 최소화할 수 있습니다. 특히 MySQL B+트리 구조에서 `루스 인덱스 스캔`(Loose Index Scan)은 GROUP BY 최적화 시 사용됩니다. 루스 인덱스 스캔은 중간에 필요하지 않은 인덱스 키는 무시하고 넘어가 불필요한 읽기 작업을 줄입니다. 주로 GROUP BY나 집합 함수(MAX, MIN)에서 해당 칼럼이 인덱스의 선두 칼럼으로 구성된 경우 적용됩니다.

ORDER BY는 인덱스의 선두 칼럼과 같은 순서로 지정된 경우 추가적인 정렬 작업이 불필요합니다.
